### PR TITLE
recommender: Initial support for third-party replica recommenders

### PR DESCRIPTION
**Pull request by iksaif**
This is a first step to implement the interface defined in DataDog/agent-payload#348

Currently this simply adds a non-implemented recommender client and the necesary logic to handle
the results. It does not allow the usage all the features made available by the recommendation
API and keep it to the minimum required to make use of it (for example simple TargetType).

Simple tests have been implemented and this was implemented to re-use as much code as possible in the replica_calculator.

Follow-up PRs will
- Provide a real implementation of the recommender
- Add more tests
- Add documentation and examples (maybe including a very simple recommender)

This PR is currently based on [lenaic/CONTINT-4412](https://github.com/DataDog/watermarkpodautoscaler/pull/221)

### CODE REVIEW

**Review by nickbirnberg — APPROVED** [review]

File: bundle/manifests/watermarkpodautoscaler.clusterserviceversion.yaml [hunk 1]
```diff
@@ -43,7 +43,7 @@ metadata:
     capabilities: Basic Install
     operators.operatorframework.io/builder: operator-sdk-v1.23.0
     operators.operatorframework.io/project_layout: go.kubebuilder.io/v3
-  name: watermarkpodautoscaler.v0.8.0-rc.2
```
[the remark this answers is not in the document]
Comment (iksaif) [reply bundle/manifests/watermarkpodautoscaler.clusterserviceversion.yaml]: fixed

File: controllers/datadoghq/replica_calculator.go [hunk 2]
```diff
@@ -237,12 +240,98 @@ func (c *ReplicaCalculator) GetResourceReplicas(logger logr.Logger, target *auto
 	return ReplicaCalculation{replicaCount, utilizationQuantity, timestamp, int32(readyPodCount), metricPos}, nil
 }
 
+// GetRecommenderReplicas contacts an external replica recommender to get the desired replica count
+func (c *ReplicaCalculator) GetRecommenderReplicas(logger logr.Logger, target *autoscalingv1.Scale, wpa *v1alpha1.WatermarkPodAutoscaler) (ReplicaCalculation, error) {
+	lbl, err := labels.Parse(target.Status.Selector)
```
[the remark this answers is not in the document]
Comment (iksaif) [reply controllers/datadoghq/replica_calculator.go:245]: fixed

[the remark this answers is not in the document]
Comment (iksaif) [reply controllers/datadoghq/replica_calculator.go:245]: (I'll have a PR to fix this just after)

**Review by vboulineau — APPROVED** [review]

**Pull request merged**